### PR TITLE
fix: regression in react-nextjs-router-push test

### PR DESCRIPTION
**Pull request by ihji**
handle unused lambda in return expression

test plan:
tested with `react-nextjs-router-push` test.

### CODE REVIEW

**Comment by ihji** [comment]
CC: @amchiclet

File: tests/rules/taint_lambda_cpp.cpp [hunk 1]
```diff
@@ -28,12 +28,10 @@ void correct_in_lambda() {
...
 
-void todo_only_sink_in_lambda(int *p) {
-    // We assume that an unused lambda is called at the end of the parent
-    // function.
+void only_sink_in_lambda(int *p) {
     source(p);
     auto f2 = [&p]() {
-        // ruleid: source-sink
+        // ok: source-sink
         sink(p);
```
Comment (IagoAbal) [inline tests/rules/taint_lambda_cpp.cpp]: Should this be an `ok` ? Is this is related to the regression ?

Comment (ihji) [reply tests/rules/taint_lambda_cpp.cpp]: I adjusted the logic slightly while addressing the regression. Now, when we place unused lambdas at the end of the parent function, they no longer have in-edges. This change also resolves the todo item in the test.

Comment (IagoAbal) [reply tests/rules/taint_lambda_cpp.cpp]: TBH when I asked to add this test I actually wanted it to be a `ruleid`, we have a lambda that when evaluated it's very likely to cause taint to go into a sink. I could be wrong but I think this deserves a separate discussion. Could we separate fixing the regression from changing the behavior here?

**Comment by IagoAbal** [comment]
If you rebase it should run the new CI check 'test-pro-rules', it would be nice to see the result here.

File: tests/rules/taint_lambda_cpp.cpp (hunk 1 above)
Comment (ihji) [reply tests/rules/taint_lambda_cpp.cpp]: done.

**Review by IagoAbal — APPROVED** [review]

File: src/analyzing/CFG_build.ml [hunk 2]
```diff
@@ -251,10 +251,16 @@ let rec cfg_stmt : state -> F.nodei option -> stmt -> cfg_stmt_result =
       add_pending_goto state newi label;
       CfgFirstLast (newi, None, false)
   | Return (tok, e) ->
-      let newi = state.g#add_node { F.n = F.NReturn (tok, e) } in
+      let new_ = F.NReturn (tok, e) in
```
Comment (IagoAbal) [inline src/analyzing/CFG_build.ml:254]: You could add a changelog too... are you following the checklist?

Comment (ihji) [reply src/analyzing/CFG_build.ml:254]: Added a changelog entry.

This is a regression fix for an unreleased change, and the original change already has a changelog. I wasn't sure if we need additional changelog for this fix.
